### PR TITLE
chore: enforce consistency in class names of demo-app

### PR DESCRIPTION
**Pull request by rafaelss95**
This:
- Ensure that all css classes are prefixed by `.demo-`;
- Removes some unused css classes;
- Enables `ViewEncapsulation` for all components, except `demo-app` since it applies CSS to the whole content;
- Organizes imports, removing some duplicates;

### CODE REVIEW

File: src/demo-app/button/button-demo.html [hunk 1]
```diff
@@ -19,64 +19,64 @@ <h4 class="section-header">Anchors</h4>
...
   <section>
     <button mat-mini-fab color="primary"><mat-icon>menu</mat-icon></button>
     <button mat-mini-fab color="accent"><mat-icon>plus_one</mat-icon></button>
     <button mat-mini-fab color="warn"><mat-icon>filter_list</mat-icon></button>
     <button mat-mini-fab disabled><mat-icon>home</mat-icon></button>
   </section>
 
-  <h4 class="section-header">Interaction/State</h4>
-  <section class="no-flex">
+  <h4 class="demo-demo-section-header">Interaction/State</h4>
```
Comment (rafaelss95) [inline src/demo-app/button/button-demo.html]: Ops, just noticed that I duplicate this. I'll fix it later.

**Comment by rafaelss95** [comment]
@crisbeto I just noticed that this is another inconsistency: there are a lot of components with disabled `VIewEncapsulation` as well as with enabled `ViewEncapsulation`. Which one is the *"correct"*?

**Comment by crisbeto** [comment]
It's fine to leave the view encapsulation on for everything @rafaelss95.

File: src/demo-app/sidenav/sidenav-demo.ts [hunk 2]
```diff
@@ -6,15 +6,15 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import {Component, ViewEncapsulation} from '@angular/core';
+import {Component } from '@angular/core';
```
Comment (crisbeto) [inline src/demo-app/sidenav/sidenav-demo.ts]: Looks like you added an extra space in here.

Comment (rafaelss95) [reply src/demo-app/sidenav/sidenav-demo.ts]: Oops, a bad replace :(

**Comment by mmalerba** [comment]
@rafaelss95 got this error while presubmitting:

```
demo-app/tabs/tabs-demo.ts(37,9): error TS4031: Public property 'tabs' of exported class has or is using private name 'DemoTab'.
```

**Pull request merged**